Added maven license plugin to pom

jhickman opened the pull request:
Updated license text to be the same formatting as the GWT 2 license content

jhickman committed:
Added maven license plugin to pom
Updated license text to be the same formatting as the GWT 2 license content

jhickman commented on pom.xml:
```diff
@@ -101,6 +101,25 @@
+          <configuration>
+            <mapping>
+              <gss>JAVADOC_STYLE</gss>
+              <java>SLASHSTAR_STYLE</java>
```
I don't have a problem with the JAVADOC_STYLE.  It's just not the style that was used before

jhickman reviewed (commented):
Running `mvn license:format` will validate the license in all the file types and update appropriately.

niloc132 commented:
Yeah, I use this plugin myself in stuff like https://github.com/niloc132/webbit-gwt/blob/master/pom.xml#L113 - though I have it set to auto-run and not bother with manually hitting it.

Want to take a bit of time and fold it into http://github.com/vertispan/some-gwt-module's setup, so that the canonical checkstyle there accepts this license format (or just doesnt check, since the license gets applied each build)? 

Might also need to futz with getting the org name to be flexible, so we have some code "Copyright Google", and some the actual author of the code, or the project, etc.

jhickman commented:
Right.   I've used this plugin in the past as well, with properties, etc.   I'll mess around with it to get it incorporated into the maven lifecycle.

jhickman commented:
Updated to have the license-plugin available to all child modules by default and added execution for process-sources lifecycle to add the license.   Also updated to include variable placeholders for year and orgName.  With this, we could add additional executions to change out the orgName for any class patterns.  gives it a bit more flexibility when it comes to applying the license.
